hybridse/src/udf/udf.cc

```diff
@@ -250,6 +251,138 @@ void timestamp_to_date(codec::Timestamp *timestamp,
 void date_to_string(codec::Date *date, hybridse::codec::StringRef *output) {
     date_format(date, "%Y-%m-%d", output);
 }
+
+/**
+* escape: null to disable escape feature
+*
+* credit:
+*  Michael Cook, https://github.com/MichaelCook/glob_match/blob/master/glob_match.cpp
+*/
+template<typename EQUAL>
+bool like_match_internal(std::string_view name,
+                         std::string_view pattern,
+                         const char* escape,
+                         EQUAL&& equal) {
+    auto it = pattern.cbegin();
+    auto end = pattern.cend();
+    auto n_it = name.cbegin();
+    auto n_end = name.cend();
+
+    while (it != end) {
+        if (n_it == n_end) {
+            return false;
+        }
+
+        char c = *it;
+        if (escape == nullptr) {
+            // escape feature is disabled
+            if (!equal(c, *n_it)) {
+                return false;
+            }
+
+            std::advance(it, 1);
+        } else if (c == *escape) {
+            // exact character match
+            if (std::next(it) == end) {
+                // the pattern is terminated with escape character, just return false
+                return false;
+            }
+            c = *std::next(it);
+            if (!equal(c, *n_it)) {
+                return false;
+            }
+
+            std::advance(it, 2);
+        } else {
+            switch (c) {
+                case '%': {
+                    std::advance(it, 1);
+                    for (auto back = n_end; back >= n_it; std::advance(back, -1)) {
+                        if (like_match_internal(std::string_view(back, std::distance(back, n_end)),
+                                                std::string_view(it, std::distance(it, end)),
+                                                escape, std::forward<EQUAL>(equal))) {
+                            return true;
+                        }
+                    }
+                    return false;
+                }
+                case '_': {
+                    break;
+                }
+                default: {
+                    if (!equal(c, *n_it)) {
+                        return false;
+                    }
+                    break;
+                }
+            }
+
+            std::advance(it, 1);
+        }
+
+        std::advance(n_it, 1);
+    }
+
+    return n_it == n_end;
+}
+
+/*
+* SQL style glob match, it use
+* - percent sign (%) as zero or more characters
+* - underscore (_) as exactly one.
+* - backslash (\) as escape character by default
+*
+* rules:
```

Comment:
The rules are quite different from what issue #224 described. I think that is because @224 borrows rules from T-SQL but the implemented rules follow MySQL's rules to some extend.
- I think T-SQL's rules are much easier to implement.
- A little concern that the pattern is default case insensitive. So what if the users try to match a string starting with 'A' exactly? I found the `regexp_match ` function but didn't find its invoker.
Update #224 description to keep consistency.

Reply:
I follow the ansi sql and postgre sql's guide as well: https://www.postgresql.org/docs/current/functions-matching.html

That detail of like predicate may different in many ways, but should easy to tweak, e.g for case sensitive just change the equal function.

bracket character is not implemented, it seems not the standard of ansi sql.

Reply:
BTW, there does has a general problem:

It may not appropriate to use backslash(\) as default escape character, since zetasql parser backslash as escape char as well. As a result, in string `'abc\n'`, `'\n'` is treated as special character, and in a like predicate, we must write `col1 LIKE 'abc\\n' ESCAPE '\\` to represent that backslash. Also, in yaml, "abc\\n" seems have different meanings.

So I'd wondering if we should tweak a lit of escape default:
1. if `ESCAPE` clause not exist, disable escape feature
2. documented: in sql, use `\\` to represent backslash
3. There is no default escape character and users are encourage to escape with another character(not escape):
    `c1 LIKE 'a$%' ESCAPE '$'`

Reply:
1. if `ESCAPE` clause not exist, using '\\' as default escape as described in https://github.com/4paradigm/OpenMLDB/discussions/686.